src/test/java/net/openhft/chronicle/queue/impl/single/StoreTailerTest.java

```diff
@@ -586,4 +586,72 @@ public void run() throws InterruptedException {
             }
         }
     }
+
+    @Test(expected = IllegalStateException.class)
+    public void cantMoveToStartDuringDocumentReading() {
+        File dir = getTmpDir();
+        try (SingleChronicleQueue queue = ChronicleQueue.singleBuilder(dir).build();
+             ExcerptTailer tailer = queue.createTailer();
+             ExcerptAppender appender = queue.createAppender()) {
+            appender.writeText("Hello World");
+            try (DocumentContext dc = tailer.readingDocument(true)) {
+                assertTrue(dc.isPresent());
+                assertTrue(dc.isMetaData());
+                assertEquals("header", dc.wire().readEvent(String.class));
+                assertTrue(tailer.toString().contains("StoreTailer{"));
+                tailer.toStart();//forbidden
+            }
+        }
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void cantMoveToEndDuringDocumentReading() {
+        File dir = getTmpDir();
+        try (SingleChronicleQueue queue = ChronicleQueue.singleBuilder(dir).build();
```

Comment:
as a general note, we should use `testBlockSize` in tests as per https://github.com/OpenHFT/Chronicle-Queue/commit/91bc0e9aac2f15f9d11bd73d9ae73ea91f41a3f2

Reply:
Added